### ISSUE TITLE
"not an allowed scheme" for GraalVM Native-Image resource:-URIs

### ISSUE THREAD

**Issue by kohlschuetter** [eclipse/jetty.project#11040]
**Jetty version(s)**
12.0.4

**Jetty Environment**
tested with ee10

**Java version/vendor** `(use: java -version)`
java version "21.0.1" 2023-10-17
Java(TM) SE Runtime Environment Oracle GraalVM 21.0.1+12.1 (build 21.0.1+12-jvmci-23.1-b19)
Java HotSpot(TM) 64-Bit Server VM Oracle GraalVM 21.0.1+12.1 (build 21.0.1+12-jvmci-23.1-b19, mixed mode, sharing)

**OS type/version**
any, macOS 14

**Description**
In [PR #9136](https://github.com/jetty/jetty.project/pull/9136), we landed on a change to support GraalVM Native-Image resource: path URIs. Unfortunately, that seems to no longer completely work on recent GraalVM versions, or maybe I'm hitting code that I haven't triggered before.

**How to reproduce?**

Try resolving `ResourceFactory.root().newResource(new URI("resource:/path/to/some/resource"))` (which is the new GraalVM resource scheme). Make sure you try this from a native-image environment.

You'll get:
```
Exception in thread "main" java.lang.IllegalArgumentException: not an allowed scheme: resource:/path/to/some/resource
	at org.eclipse.jetty.util.resource.PathResource.<init>(PathResource.java:168)
	at org.eclipse.jetty.util.resource.PathResourceFactory.newResource(PathResourceFactory.java:29)
	at org.eclipse.jetty.util.resource.ResourceFactoryInternals$CompositeResourceFactory.newResource(ResourceFactoryInternals.java:232)
...
```

The reason for this exception is that `PathResource` has its own set of allowed URI schemes, independent of what is defined in `ResourceFactoryInternals`. Our previous workaround no longer fully works because [Graal issue 5720](https://github.com/oracle/graal/issues/5720) was fixed, so we no longer use our workaround `GraalIssue5720PathResourceFactory`.

A proposed fix is to register the `resource:` scheme from `ResourceFactoryInternals` if we detect we're running in native-image.

**Comment by joakime** [comment]
@kohlschuetter can you see if PR #11057 is enough for you?

**Comment by kohlschuetter** [comment]
@joakime Confirming that PR #11057 fixes this issue. Thanks!

**Comment by joakime** [comment]
Merged PR #11057
